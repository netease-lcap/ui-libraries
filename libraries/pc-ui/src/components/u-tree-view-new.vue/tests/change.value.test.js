import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import { UTreeViewNew } from '../index';

test('test change value', async () => {
  let syncValue = '';
  const DefaultValue = ['节点 1', '节点 2'];
  const wrapper = mount(UTreeViewNew, {
    propsData: {
      value: DefaultValue,
      valueField: 'text1',
      textField: 'text1',
      checkable: true,
      dataSource: [
        {
          text1: '节点 1',
          children: [
            { text1: '节点 1.1' },
            {
              text1: '节点 1.2',
              children: [{ text1: '节点 1.2.1' }, { text1: '节点 1.2.2' }],
            },
            { text1: '节点 1.3' },
            { text1: '节点 1.4' },
          ],
        },
        { text1: '节点 2' },
        {
          text1: '节点 3',
          children: [{ text1: '节点 3.1' }, { text1: '节点 3.2' }],
        },
      ],
    },
    listeners: {
      'update:value': (v) => {
        syncValue = v;
      },
    },
  });

  await wrapper.vm.$nextTick();
  expect(syncValue).toEqual(['节点 2', '节点 1.1', '节点 1.2.1', '节点 1.2.2', '节点 1.3', '节点 1.4']);

  await wrapper.setProps({
    value: ['节点 2', '节点 3', '节点 1.2'],
  });

  await wrapper.vm.$nextTick();
  expect(syncValue).toEqual(['节点 2', '节点 1.2.1', '节点 1.2.2', '节点 3.1', '节点 3.2']);
});

test('test checked controlled value', async () => {
  let syncValue = '';
  const DefaultValue = ['节点 1', '节点 2'];
  const wrapper = mount(UTreeViewNew, {
    propsData: {
      value: DefaultValue,
      valueField: 'text1',
      textField: 'text1',
      checkable: true,
      checkControlled: true,
      dataSource: [
        {
          text1: '节点 1',
          children: [
            { text1: '节点 1.1' },
            {
              text1: '节点 1.2',
              children: [{ text1: '节点 1.2.1' }, { text1: '节点 1.2.2' }],
            },
            { text1: '节点 1.3' },
            { text1: '节点 1.4' },
          ],
        },
        { text1: '节点 2' },
        {
          text1: '节点 3',
          children: [{ text1: '节点 3.1' }, { text1: '节点 3.2' }],
        },
      ],
    },
    listeners: {
      'update:value': (v) => {
        syncValue = v;
      },
    },
  });

  await wrapper.vm.$nextTick();
  expect(wrapper.vm.currentValues).toEqual(DefaultValue);

  await wrapper.setProps({
    value: ['节点 2', '节点 3', '节点 1.2'],
  });

  await wrapper.vm.$nextTick();
  expect(wrapper.vm.currentValues).toEqual(['节点 2', '节点 3', '节点 1.2']);
  expect(syncValue).toBe('');
});
