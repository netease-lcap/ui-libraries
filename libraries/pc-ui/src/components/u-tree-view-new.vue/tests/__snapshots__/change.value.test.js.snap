// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test change dataSource 1`] = `
<div>
  <div class="root">
    <div class="root">
      <div class="item" style="padding-left: 0px;" tabindex="0">
        <div class="expander" expand-trigger="click" expanded="true" style="width: 30px;"></div>
        <div class="text" draggable="false" style="margin-left: 30px;">
          <label class="root" show-title="true" tabindex="0">
            <span class="box" status="null"></span>
            <span vusion-slot-name="item"></span>
          </label>
          <span vusion-slot-name="item">节点 1</span>
        </div>
      </div>
      <div class="sub">
        <div class="root">
          <div class="item" style="padding-left: 30px;" tabindex="0">
            <div class="text" draggable="false" style="margin-left: 30px;">
              <label class="root" show-title="true" tabindex="0">
                <span class="box" status="false"></span>
                <span vusion-slot-name="item"></span>
              </label>
              <span vusion-slot-name="item">节点 1.1</span>
            </div>
          </div>
          <div class="sub" style="display: none;">
          </div>
        </div>
        <div class="root">
          <div class="item" style="padding-left: 30px;" tabindex="0">
            <div class="expander" expand-trigger="click" expanded="true" style="width: 30px;"></div>
            <div class="text" draggable="false" style="margin-left: 30px;">
              <label class="root" show-title="true" tabindex="0">
                <span class="box" status="false"></span>
                <span vusion-slot-name="item"></span>
              </label>
              <span vusion-slot-name="item">节点 1.2</span>
            </div>
          </div>
          <div class="sub">
            <div class="root">
              <div class="item" style="padding-left: 60px;" tabindex="0">
                <div class="text" draggable="false" style="margin-left: 30px;">
                  <label class="root" show-title="true" tabindex="0">
                    <span class="box" status="false"></span>
                    <span vusion-slot-name="item"></span>
                  </label>
                  <span vusion-slot-name="item">节点 1.2.1</span>
                </div>
              </div>
              <div class="sub" style="display: none;">
              </div>
            </div>
            <div class="root">
              <div class="item" style="padding-left: 60px;" tabindex="0">
                <div class="text" draggable="false" style="margin-left: 30px;">
                  <label class="root" show-title="true" tabindex="0">
                    <span class="box" status="false"></span>
                    <span vusion-slot-name="item"></span>
                  </label>
                  <span vusion-slot-name="item">节点 1.2.2</span>
                </div>
              </div>
              <div class="sub" style="display: none;">
              </div>
            </div>
          </div>
        </div>
        <div class="root">
          <div class="item" style="padding-left: 30px;" tabindex="0">
            <div class="text" draggable="false" style="margin-left: 30px;">
              <label class="root" show-title="true" tabindex="0">
                <span class="box" status="true"></span>
                <span vusion-slot-name="item"></span>
              </label>
              <span vusion-slot-name="item">节点 1.3</span>
            </div>
          </div>
          <div class="sub" style="display: none;">
          </div>
        </div>
        <div class="root">
          <div class="item" style="padding-left: 30px;" tabindex="0">
            <div class="text" draggable="false" style="margin-left: 30px;">
              <label class="root" show-title="true" tabindex="0">
                <span class="box" status="true"></span>
                <span vusion-slot-name="item"></span>
              </label>
              <span vusion-slot-name="item">节点 1.4</span>
            </div>
          </div>
          <div class="sub" style="display: none;">
          </div>
        </div>
      </div>
    </div>
    <div class="root">
      <div class="item" style="padding-left: 0px;" tabindex="0">
        <div class="text" draggable="false" style="margin-left: 30px;">
          <label class="root" show-title="true" tabindex="0">
            <span class="box" status="false"></span>
            <span vusion-slot-name="item"></span>
          </label>
          <span vusion-slot-name="item">节点 2</span>
        </div>
      </div>
      <div class="sub" style="display: none;">
      </div>
    </div>
    <div class="root">
      <div class="item" style="padding-left: 0px;" tabindex="0">
        <div class="text" draggable="false" style="margin-left: 30px;">
          <label class="root" show-title="true" tabindex="0">
            <span class="box" status="false"></span>
            <span vusion-slot-name="item"></span>
          </label>
          <span vusion-slot-name="item">节点 3</span>
        </div>
      </div>
      <div class="sub" style="display: none;">
      </div>
    </div>
  </div>
</div>
`;

exports[`test change dataSource 2`] = `
<div>
  <div class="root">
    <div class="root">
      <div class="item" style="padding-left: 0px;" tabindex="0">
        <div class="expander" expand-trigger="click" expanded="true" style="width: 30px;"></div>
        <div class="text" draggable="false" style="margin-left: 30px;">
          <label class="root" show-title="true" tabindex="0">
            <span class="box" status="true"></span>
            <span vusion-slot-name="item"></span>
          </label>
          <span vusion-slot-name="item">节点 1</span>
        </div>
      </div>
      <div class="sub">
        <div class="root">
          <div class="item" disabled="disabled" style="padding-left: 30px;" tabindex>
            <div class="text" draggable="false" style="margin-left: 30px;">
              <label class="root" disabled="disabled" show-title="true" tabindex="0">
                <span class="box" disabled="disabled" status="false"></span>
                <span vusion-slot-name="item"></span>
              </label>
              <span vusion-slot-name="item">节点 1.1</span>
            </div>
          </div>
          <div class="sub" style="display: none;">
          </div>
        </div>
        <div class="root">
          <div class="item" disabled="disabled" style="padding-left: 30px;" tabindex>
            <div class="expander" expand-trigger="click" expanded="true" style="width: 30px;"></div>
            <div class="text" draggable="false" style="margin-left: 30px;">
              <label class="root" disabled="disabled" show-title="true" tabindex="0">
                <span class="box" disabled="disabled" status="false"></span>
                <span vusion-slot-name="item"></span>
              </label>
              <span vusion-slot-name="item">节点 1.2</span>
            </div>
          </div>
          <div class="sub">
            <div class="root">
              <div class="item" disabled="disabled" style="padding-left: 60px;" tabindex="0">
                <div class="text" draggable="false" style="margin-left: 30px;">
                  <label class="root" disabled="disabled" show-title="true" tabindex="0">
                    <span class="box" disabled="disabled" status="false"></span>
                    <span vusion-slot-name="item"></span>
                  </label>
                  <span vusion-slot-name="item">节点 1.2.1</span>
                </div>
              </div>
              <div class="sub" style="display: none;">
              </div>
            </div>
            <div class="root">
              <div class="item" disabled="disabled" style="padding-left: 60px;" tabindex="0">
                <div class="text" draggable="false" style="margin-left: 30px;">
                  <label class="root" disabled="disabled" show-title="true" tabindex="0">
                    <span class="box" disabled="disabled" status="false"></span>
                    <span vusion-slot-name="item"></span>
                  </label>
                  <span vusion-slot-name="item">节点 1.2.2</span>
                </div>
              </div>
              <div class="sub" style="display: none;">
              </div>
            </div>
          </div>
        </div>
        <div class="root">
          <div class="item" style="padding-left: 30px;" tabindex="0">
            <div class="text" draggable="false" style="margin-left: 30px;">
              <label class="root" show-title="true" tabindex="0">
                <span class="box" status="true"></span>
                <span vusion-slot-name="item"></span>
              </label>
              <span vusion-slot-name="item">节点 1.3</span>
            </div>
          </div>
          <div class="sub" style="display: none;">
          </div>
        </div>
        <div class="root">
          <div class="item" style="padding-left: 30px;" tabindex="0">
            <div class="text" draggable="false" style="margin-left: 30px;">
              <label class="root" show-title="true" tabindex="0">
                <span class="box" status="true"></span>
                <span vusion-slot-name="item"></span>
              </label>
              <span vusion-slot-name="item">节点 1.4</span>
            </div>
          </div>
          <div class="sub" style="display: none;">
          </div>
        </div>
      </div>
    </div>
    <div class="root">
      <div class="item" style="padding-left: 0px;" tabindex="0">
        <div class="text" draggable="false" style="margin-left: 30px;">
          <label class="root" show-title="true" tabindex="0">
            <span class="box" status="false"></span>
            <span vusion-slot-name="item"></span>
          </label>
          <span vusion-slot-name="item">节点 2</span>
        </div>
      </div>
      <div class="sub" style="display: none;">
      </div>
    </div>
    <div class="root">
      <div class="item" style="padding-left: 0px;" tabindex="0">
        <div class="text" draggable="false" style="margin-left: 30px;">
          <label class="root" show-title="true" tabindex="0">
            <span class="box" status="false"></span>
            <span vusion-slot-name="item"></span>
          </label>
          <span vusion-slot-name="item">节点 3</span>
        </div>
      </div>
      <div class="sub" style="display: none;">
      </div>
    </div>
  </div>
</div>
`;
